### PR TITLE
Hotfix/contact selection

### PR DESCRIPTION
**Pull request by bortoletti-giorgia**
## Premise

Now user can select a contact and have its phone number in the home screen.
The item's background changes color and the contact is saved in the `MainActivity`. So when home screen is resumed the EditText with phone is set with phone number of selected contact.
When you re-open contacts list, previous selected contact remains highlighted.
If you click on it and come back to home, the EditText returns empty.
_For now, to go back to the home you have to press the back button. I haven't added any buttons or timers._

## Changes

- Set a Listener on contact Item (`ContactViewHolder`).
- Add some static methods in the `MainActivity` to set selected contact from `SMSContactRecylerAdapter`.

### Footer

After merging PR of Riccardo, I'll update this code in the Fragments.

### CODE REVIEW

File: app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java [hunk 1]
```diff
@@ -20,6 +20,9 @@
 import ingsw.group1.findmyphone.Manager;
 import ingsw.group1.findmyphone.R;
 import ingsw.group1.findmyphone.ServiceManager;
+import ingsw.group1.findmyphone.contacts.GenericContact;
+import ingsw.group1.findmyphone.contacts.SMSContact;
+import ingsw.group1.findmyphone.contacts.SMSContactRecyclerAdapter;
```
Comment (RicDeZen) [inline app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java]: Codacy Issue found: [Unused import - ingsw.group1.findmyphone.contacts.SMSContactRecyclerAdapter.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5090367)

**Comment by RicDeZen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5090367&bid=16745592)

File: app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java [hunk 2]
```diff
@@ -20,6 +20,9 @@
 import ingsw.group1.findmyphone.Manager;
 import ingsw.group1.findmyphone.R;
 import ingsw.group1.findmyphone.ServiceManager;
+import ingsw.group1.findmyphone.contacts.GenericContact;
```
Comment (RicDeZen) [inline app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java]: Codacy Issue found: [Unused import - ingsw.group1.findmyphone.contacts.GenericContact.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5090367)

File: app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java (hunk 1 above)
Comment (RicDeZen) [inline app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java]: Codacy Issue found: [Avoid unused imports such as 'ingsw.group1.findmyphone.contacts.SMSContactRecyclerAdapter'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5090367)

File: app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java (hunk 2 above)
Comment (RicDeZen) [inline app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java]: Codacy Issue found: [Avoid unused imports such as 'ingsw.group1.findmyphone.contacts.GenericContact'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5090367)

**Comment by RicDeZen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5090367&bid=16745592)

**Comment by ricdezen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactRecyclerAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5090367&bid=16745592)

**Pull request closed without merging**